index.mjs

```diff
@@ -6,21 +6,41 @@ import http from 'http';
 import { exec } from 'child_process';
 import os from 'os';
 import net from 'net';
+import pkg from 'enquirer';
+const { prompt } = pkg;
 
 // Default port for the server
 let DEFAULT_PORT = 9666;
 
```

Comment:
I would refactor it as follows (Or something similar):
```js
const params = [
  {
    type: 'input',
    name: 'repoUrl',
    message: 'Enter the url of the repo to analyze',
    required: true
  },
  {
    type: 'input',
    name: 'daysAmount',
    message: 'Enter the number of days to consider',
    required: true
  },
  {
    type: 'input',
    name: 'branchName',
    message: 'Enter the name of the branch to consider',
    initial: 'main',
    required: true
  },
  {
    type: 'select',
    name: 'outputFormat',
    message: 'Choose the output format',
    choices: ['json', 'html'],
    initial: 'json',
    required: true
  }
];

let useEnquirer = true;
if(process.argv.length === params.length + 2)
  useEnquirer = false;

let repoUrl, daysAmount, branchName, outputFormat;

if(useEnquirer) {
  const response = await prompt(params);
  repoUrl = response.repoUrl;
  daysAmount = response.daysAmount;
  branchName = response.branchName;
  outputFormat = response.outputFormat;
} else {
  const response = process.argv.slice(2);
  repoUrl = response[0] || params.find(p => p.name === 'repoUrl').initial || '';
  daysAmount = response[1] || params.find(p => p.name === 'daysAmount').initial || '';
  branchName = response[2] || params.find(p => p.name === 'branchName').initial || '';
  outputFormat = response[3] || params.find(p => p.name === 'outputFormat').initial || '';
}
```

So we can have a single configuration spreaded across both needs